system/jlib/jptree.cpp

```diff
@@ -3057,8 +3182,10 @@ bool CAtomPTree::removeAttribute(const char *key)
         return false;
     numAttrs--;
     CriticalBlock block(hashcrit);
-    attrHT->removekey(del->key, isnocase());
-    attrHT->removeval(del->value);
+    if (del->key.isPtr())
+        attrHT->removekey(del->key.getPtr(), isnocase());
+    if (del->value.isPtr())
+        attrHT->removeval(del->value.getPtr());
     AttrValue *newattrs = newAttrArray(numAttrs);
```

Comment:
future:  Reallocating on each removal is potentially painful - especially when multiple attributes are being removed (e.g synchronizePtree), or if there is a chain of setProp/removeProp.